authorization-common/src/main/java/com/sequenceiq/authorization/service/ResourceAuthorizationService.java

```diff
@@ -64,7 +68,7 @@ public void authorize(String userCrn, ProceedingJoinPoint proceedingJoinPoint, M
                     LOGGER.debug("Resource authorization failed: {}", failedAuthorization.toString(rightMapper));
                 }
                 if (authzEntitled) {
-                    throw new AccessDeniedException(failedAuthorization.getAsFailureMessage(rightMapper));
+                    throw new AccessDeniedException(failedAuthorization.getAsFailureMessage(rightMapper, nameMapper));
```

Comment:
cc: @lnardai , @sodre90 

This structure can be quite inefficient when we have a bunch of resources and the user doesn't have right on most of them (delete 100 datahubs and the user doesn't have rights on them maps to 101 different database queries, 1 for the env and 100 for the datahubs). Basically the underlying behaviour is like you do this:

```
for (String crn : crns) {
  String name = db.query("SELECT name FROM xy WHERE crn = :crn")
}
```

instead of `SELECT name FROM xy WHERE crn IN (:crns)` with all the .

I would restructure the logic based on the followings:

- collect the crns from the failedAuthorization to a `Set<String>`,
- sort them into a `Map<ResourceType, Set<String>>`, basically crns by resource types,
- change the `String ResourceNameProvider.getNameByCrn(String)` to `Map<String, String> ResourceNameProvider.getNamesByCrns(Set<String>)` and use the `IN` sql stament in the underlying implementations (most of the time this is easily achiavable except `EnvironmentClientService`),
- pass the result `Map<String, String>` name-crn map to the `AuthorizationRule`.

I checked the code, the above solution wouldn't complicate the code at all and we wouldn't have worse and worse performance with increasing request size. I can even give a few ours to implement this and compare the solution if you wan't it.

Reply:
1) I don't think that this would cause performance issues. The names only collected in the case of failed authorizations (in most cases there is only _one_ crn)
2) the code would be more complicated, and different form the already implemented logic (e.g. rightMapper)